olive/resource_path.py

```diff
@@ -344,6 +350,66 @@ def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: b
         return str(new_path)
 
 
+class AzureMLRegistryModel(ResourcePath):
```

Comment:
since `save_to_dir` for this class is very similar to that of `AzureMLModel`, could you abstract the logic to avoid duplicate code? Perhaps using a common base classs or helper function for download?

Reply:
Done.